JBIDE-20152 scripts for tagging via buildinfo

maxandersen opened the pull request:
Seperate python utilities to convert buildinfo.json to csv and then one for making the tags.

```
curl http://download.jboss.org/jbosstools/static/mars/development/updates/core/4.3.0.Beta1/buildinfo.json | python buildinfo_to_tags.py -n jbosstools-4.3.0.Beta1 | python tagrepos.py maxandersen secret
```

nickboldt commented:
+0.5 for buildinfo_to_tags.py -- you need to document usage to include the mention of curl on lines 12-14 and line 24. 

`curl http://download.jboss.org/jbosstools/static/mars/development/updates/core/4.3.0.Beta1/buildinfo.json | python buildinfo_to_tags.py -n jbosstools-4.3.0.Beta1`

After that, +1. 

Did not review  util/checktags.py, but I assume you tested it because you found missing comma on line 47/59.

maxandersen commented:
Was in a hurry so I missed that I included my updates to checktags.py in this PR. I think I'll split those out.

I have a hard time following your comments that just list line numbers and I can't really see why/what in those lines you mean curl should be mentioned ? Could you put the comments on the lines so I can get a better idea what you refer to ?

And yes, I'll go and add usage text to these.

maxandersen commented:
Updated the branch. 

checktags.py was unrelated to this thus moved to separate PR #133

tagrepos.py was missing from this PR, that is now fixed so you can actually run the commands.

readme.adoc updated with description of the new scripts.

nickboldt commented on util/buildinfototags.py:
```diff
@@ -0,0 +1,44 @@
+# moar output, set = 1
+debug=0
+
+usage = "Usage: %prog [-n NAME] \n\n\
```
you need to document usage to include the mention of curl on lines 12-14 and line 24.

maxandersen replied:
not sure why you want curl. why not cat or anything else ? 

I'll update it to explicitly state it needs its input on standard input.

nickboldt commented on util/buildinfototags.py:
```diff
@@ -0,0 +1,44 @@
+if (not options.name):
+    parser.error("Must to specify ALL commandline flags; use -h for help")
+
+j = json.load(sys.stdin) 
```
you need to document usage to include the mention of curl on lines 12-14 and line 24.

maxandersen replied:
not following what you expect here, besides what already mentioned above?

nickboldt replied:
How about...
    # read from standard input, eg., cat buildinfo.json | python buildinfo2tags.py or 
    # curl http://path/to/buildinfo.json | python buildinfo_to_tags.py

nickboldt commented on util/tagrepos.py:
```diff
@@ -0,0 +1,46 @@
+    print "jbosstools/jbosstools-hibernate, a1f287bf825a891901b41739f6a4bbaafc70bcdd, jbosstools-4.2.1.Alpha23"
+    print
+    print "Useful when combined with buildinfo.json: "
+    print "cat buildinfo.json | python buildinfo_to_tags.py -n <tagname> | python tagrepos.py <taggeruser> <secret>" 
```
I link this filename better: "buildinfo_to_tags" is more easy to read than "buildinfototags" which looks like "build in foto tags". However, you need to then rename the file in this PR: https://github.com/maxandersen/jbosstools-build-ci/blob/tagtools/util/buildinfototags.py

At any rate, please be consitent w/ your usage instructions and filenames.

maxandersen replied:
that would be introducing yet another naming convention used in this repo. not going there. 

I'l try come up with simpler name.

nickboldt commented on util/tagrepos.py:
```diff
@@ -0,0 +1,46 @@
+    print "cat buildinfo.json | python buildinfo_to_tags.py -n <tagname> | python tagrepos.py <taggeruser> <secret>" 
+    sys.exit(-1)
+
+repos = csv.reader(sys.stdin)
```
you need to document usage to include the mention of the fact that you're simply piping the file into python instead of specifying it on the commandline. Incidentally, why do you feel a pipe is easier to use than "-f filename" ? Just curious.

maxandersen replied:
because it requires less code to handle and it can then be combined.

The usage already mention piping is needed.

nickboldt replied:
Why split these into two scripts when they're kind of useless when separated? Why would you generate the csv file and then not use it? Or, what good would the tagrepos.py script be without the csv file as input?

maxandersen replied:
the csv can be generated as part of release cycle and discover when errors like missing tags (see freemarker and portal) and then passed on to script or whoever will do the tagging.

also - code is much simpler not having to handle buildinfo.json.

nickboldt commented on util/tagrepos.py:
```diff
@@ -0,0 +1,46 @@
+
+if len(sys.argv) <> 3:
+    print "Please specify both username and github password."
+    print "Usage: tagrepos.py <username> <password>"
```
you need to document usage to include the mention of the fact that you're simply piping the file into python instead of specifying it on the commandline.

maxandersen replied:
what do you suggest besides all the examples below ?

nickboldt replied:
"Useful when combined?" More like "pointless unless combined." :D 

Maybe...

     + print "Usage: cat somefile.csv | tagrepos.py <username> <password>" 

?

maxandersen replied:
added cat to usage txt.

nickboldt commented on util/tagrepos.py:
```diff
@@ -0,0 +1,46 @@
+        if ge.status == 422:
+            results = str(ge)
+        elif ge.status == 400:
+            results = "Bad things happened 400"
```
Bad things happened? Can't you capture the actual error here for a more meaningful result?

maxandersen replied:
the main reason for this is that gihtub error messages are less than great if the issue is related to missing resources.

But did not want everything to just fail because 1 out of N repos failed with "known" errors like 400/422. Thus i've updated to just do what  422 error did and print the raw error.

nickboldt commented on util/tagrepos.py:
```diff
@@ -0,0 +1,46 @@
+    except GithubException as ge:
+        if ge.status == 422:
+            results = str(ge)
+        elif ge.status == 400:
```
Since both ifs now do the same thing, could the y be combined into a single line?

    if ge.status == 400 or ge.status == 422:
        results = str(ge)

maxandersen replied:
yes, but that is just going to make it harder to edit/debug. But i've changed it since not major deal.
